service/hook/bitbucketserver/bitbucketserver_test.go

```diff
@@ -1401,6 +1558,33 @@ func Test_HookProvider_TransformRequest(t *testing.T) {
 		require.Equal(t, false, hookTransformResult.DontWaitForTriggerResponse)
 	}
 
+	t.Log("Test with Sample Pull Request approved Data")
+	{
+		request := http.Request{
+			Header: http.Header{
+				"X-Event-Key":  {"pr:reviewer:approved"},
+				"Content-Type": {"application/json; charset=utf-8"},
+			},
+			Body: ioutil.NopCloser(strings.NewReader(samplePullRequestApprovedData)),
+		}
+		hookTransformResult := provider.TransformRequest(&request)
+		require.NoError(t, hookTransformResult.Error)
+		require.False(t, hookTransformResult.ShouldSkip)
+		require.Equal(t, []bitriseapi.TriggerAPIParamsModel{
+			{
+				BuildParams: bitriseapi.BuildParamsModel{
```

Comment:
Note: with the current implementation the `approved` event will call the Bitrise Build Trigger API with the same parameters as the regular push/pull-request events.

This means that the Trigger API will have no information (no parameter passed) indicating that this was an approve event, and so no filtering can be applied on it.

Reply:
Hi @viktorbenei,

I am aware that this will indeed be the case. From my perspective, the pr:approved event is also a pull-request event coming from Bitbucket Server so I presume we can consider this as equal in Bitrise.

If this is blocking for this PR, I can try to have another look at it, but some directions would come in handy as I'm not too familiar with both Go as this codebase.

Reply:
Any updates on this?